api/spec/features/checkout_spec.rb

```diff
@@ -61,6 +61,8 @@ def create_line_item(variant, quantity = 1)
     end
 
     def add_promotion(_promotion)
+      expect(Spree::Deprecation).to receive(:warn)
```

Comment:
what about use the new api endpoint in the spec instead of keeping the old one and expecting a deprecation warning?

Reply:
Sure!